### DISCUSSION TITLE
Déploiement sur infomaniak (serveur cloud dédié)

### DISCUSSION THREAD

**Discussion by KevinLedoyen** [strapi/strapi#7628] [Legacy Topics]
**EDIT by @derrickmehaffy* In the future please ensure you only use english, thank you!

------ EN
Hello to all, 

We can't manage to launch the strapi application, which we developed, on our dedicated cloud server on infomaniak : 

- The npm start function (after an npm install and an npm build, does not work)
After several tests, here are the results: 

.env 
NODE_ENV=production
HOST=localhost
PORT=1337

=> npm start : OK it starts, we go to the preview site (domain name being transferred to the server but it should not have any impact), and the preview.infomaniak.com:1337 doesn't give anything (I guess it's normal, the server only listens to port 80 and 443) EDIT : if we open port 1337, preview.infomaniak.com:1337 doesn't give anything.

So another try with

.env 
NODE_ENV=production
HOST=localhost
PORT=80

=> npm start (and npm build before for the form) : it crashes and gives the error below

    error Error: listen EACCES: permission denied 127.0.0.1:80
    at Server.setupListenHandle [as _listen2] (net.js:1262:19)
    at listenInCluster (net.js:1327:12)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1460:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:61:10)

We want our app to work with port 80 and 443, how to do?

**Comment by derrickmehaffy** [comment]
> HOST=localhost

This is most likely your issue, when this is set to local host it will **only listen** on localhost, if you want the Strapi application to be accessable outside the system either you need to change this to `0.0.0.0` or have a proxy application like Nginx ([Doc guide](https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html) ) or HAProxy ([Doc guide](https://strapi.io/documentation/v3.x/deployment/haproxy-proxy.html) ) that will proxy the requests from port 80 to localhost:1337,

> PORT=80

On linux systems to bind to any port below 1024 you need to be a root user and running a node application as root is **extremely insecure** and I would strong advise against that, in fact directly exposing a Strapi application to the public web is also highly cautioned against, or just node in general. You really should run it behind a proxy application.

> We want our app to work with port 80 and 443, how to do?

As mentioned above you should use a proxy application to handle the "SSL offload" and let Nginx handle the SSL certificate that will proxy all the requests back to Strapi, for this reason Strapi does not natively support SSL at all due to how node functions and updating the SSL cert requiring the server to restart (causing downtime) as well as difficulty scaling Strapi to meet load properly. Allowing a proxy application to handle this removes these concerns.

**Reply by KevinLedoyen** [reply]
Thanks a lot for your answer ! 👍 

With 0.0.0.0, port 1337, it start and we can access on mypreview.infomaniak.com:1337 (width port 1337 open)

For now, the domain is being transferred so i cannot install SSL, so i cannot configure HaProxy... So I will update this post later to let you know ! :)
I guess I will have to close port 1337 for security after

Thanks again !

**Reply by derrickmehaffy** [reply]
No problem
